workflows/illumina.nf

```diff
@@ -107,16 +166,102 @@ workflow ILLUMINA {
+    .map { it[1] }
+    .splitCsv(header: false, sep:",")
+    .map{ [it[0], it[1], it[2]] }
+    .combine(CAT_ILLUMINA_FASTQ.out.reads.map { [it[0].id, it[1]] }, by: 0)
+    .set { ch_sample_segment } 
+
+  // Pull segment reference sequence for each sample
+  SEQTK_SEQ(ch_sample_segment, ZSTD_DECOMPRESS_FASTA.out.file)
+  ch_versions = ch_versions.mix(SEQTK_SEQ.out.versions)
+
+  // Map reads against segment reference sequences using Minimap2
+  MINIMAP2(SEQTK_SEQ.out.sample_info)
+  ch_versions = ch_versions.mix(MINIMAP2.out.versions)
+
+  // Generate coverage and stats
+  MOSDEPTH_GENOME(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(MOSDEPTH_GENOME.out.versions)
+
+  FREEBAYES(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(FREEBAYES.out.versions)
+
+  BCF_FILTER_FREEBAYES(FREEBAYES.out.vcf, params.major_allele_fraction)
+  ch_versions = ch_versions.mix(BCF_FILTER_FREEBAYES.out.versions)
+  ch_vcf_filter = BCF_FILTER_FREEBAYES.out.vcf
+
+  VCF_FILTER_FRAMESHIFT(ch_vcf_filter)
+  ch_versions = ch_versions.mix(VCF_FILTER_FRAMESHIFT.out.versions)
+
+  BCFTOOLS_STATS(VCF_FILTER_FRAMESHIFT.out.vcf)
+  ch_versions = ch_versions.mix(BCFTOOLS_STATS.out.versions)
+
+  VCF_FILTER_FRAMESHIFT.out.vcf
+    .combine(MOSDEPTH_GENOME.out.bedgz, by: [0, 1, 2]) // combine channels based on sample_name, segment and accession_id
+    .set { ch_bcf_consensus } // ch_bcf_consensus: [sample_name, segment, id, fasta, filt_vcf, mosdepth_per_base]
+
+  COVERAGE_PLOT(ch_bcf_consensus, params.low_coverage)
+  ch_versions = ch_versions.mix(COVERAGE_PLOT.out.versions)
+
+  // Generate consensus sequences
+  BCF_CONSENSUS(ch_bcf_consensus, params.low_coverage)
+  ch_versions = ch_versions.mix(BCF_CONSENSUS.out.versions)
+
+  BCF_CONSENSUS.out.fasta
+    .groupTuple(by: 0)
+    .set { ch_final_consensus }
+
+  CAT_CONSENSUS(ch_final_consensus)
+  ch_versions = ch_versions.mix(CAT_CONSENSUS.out.versions)
+
+  CAT_CONSENSUS.out.fasta
+    .map { [[id:it[0]], it[1]] }
+    .set { ch_cat_consensus }
+
+  BLAST_BLASTN_CONSENSUS(ch_cat_consensus, BLAST_MAKEBLASTDB_NCBI.out.db)
+  ch_versions = ch_versions.mix(BLAST_BLASTN_CONSENSUS.out.versions)
+
+  ch_blastn_consensus = BLAST_BLASTN_CONSENSUS.out.txt.collect({ it[1] })
+  SUBTYPING_REPORT_BCF_CONSENSUS(
+    ZSTD_DECOMPRESS_CSV.out.file, 
+    ch_blastn_consensus,
+    CHECK_SAMPLE_SHEET.out
+  )
+  ch_versions = ch_versions.mix(SUBTYPING_REPORT_BCF_CONSENSUS.out.versions)
+
+  workflow_summary    = Schema.params_summary_multiqc(workflow, summary_params)
+  ch_workflow_summary = Channel.value(workflow_summary)
+  ch_multiqc_config = Channel.fromPath("$projectDir/assets/multiqc_config.yaml")
 
   MQC_VERSIONS_TABLE(ch_versions.unique().collectFile(name: 'collated_versions.yml'))
+
+  // MultiQC
+  ch_workflow_summary = Channel.value(Schema.params_summary_multiqc(workflow, summary_params))
+  ch_multiqc_config = Channel.fromPath("$projectDir/assets/multiqc_config.yaml")
+
+  MULTIQC(
```

Comment:
MultiQC report looks great! It'd be useful to have the `multiqc_report.html` uploaded as an artifact from the Illumina workflow CI testing similar to the Nanopore workflow:

https://github.com/CFIA-NCFAD/nf-flu/blob/4deac8b8d1b0db37277aae8c9190bf18fcbc7145/.github/workflows/ci.yml#L254-L259

Reply:
@peterk87 Added artifact for the Illumina workflow. Please approve testing to see if the artifact is generated to resolve.

Reply:
MultiQC artifact added.